dpc/src/block/transactions.rs

```diff
@@ -176,6 +176,14 @@ impl<N: Network> Transactions<N> {
     ) -> Result<MerklePath<N::TransactionsRootParameters>> {
         Ok(self.tree.generate_proof(index, &leaf)?)
     }
+
+    /// Returns records from the transactions belonging to the given account view key.
+    pub fn to_decrypted_records(&self, account_view_key: &ViewKey<N>) -> Vec<Record<N>> {
+        self.transactions
+            .iter()
+            .flat_map(|transaction| transaction.to_decrypted_records(account_view_key))
+            .collect()
```

Comment:
Instead of collecting, can you return this as an `impl Iterator`?

(ditto for `Transaction::to_decrypted_records` if it is not an iterator already)

You will see significant performance gains in what you're doing.

Reply:
Yes I can remove the '.collect' calls.

Thoughts on calling '.into_par_iter'?

Reply:
If you can make it a conditional, maybe for now set it to around 500. So if its >500 use par_iter, if its below, use iter.

Reply:
@collinc97